pkg/oauth/tokenrequest/request_token.go

```diff
@@ -55,12 +59,44 @@ type RequestTokenOptions struct {
 	OsinConfig   *osincli.ClientConfig
 	Issuer       string
 	TokenFlow    bool
+
+	AuthorizationURLHandler AuthorizationURLHandlerFunc
+	LocalCallbackServer     *callbackServer
 }
 
+// AuthorizationURLHandlerFunc defines how the authorization URL of the OAuth Code Grant flow
+// should be handled; for example use this function to take the URL and open it in a browser.
+type AuthorizationURLHandlerFunc func(url *url.URL) error
+
 // RequestToken uses the cmd arguments to locate an openshift oauth server and attempts to authenticate via an
 // OAuth code flow and challenge handling.  It returns the access token if it gets one or an error if it does not.
 func RequestToken(clientCfg *restclient.Config, challengeHandlers ...challengehandlers.ChallengeHandler) (string, error) {
```

Comment:
@stlaz would it make sense to rename this func and keep the naming convention `*WithChallengers` `*WithLocalCallback` here as well? In the same spirit to rename `RequestTokenWithAuthorization` to match the convention too.

Or would it be too much of a breaking change you reckon?

Reply:
Proposal can be seen here: https://github.com/openshift/library-go/pull/1531/files#diff-f30259f7a13841a226555f2801678f096da8bb80456cf414201b8cfd647da728R73-R80

Reply:
... and this is what it would look like from the client side:

```go
var token string
var err error
if o.WebLogin {
	urlHandler := func(u *url.URL) error {...}
	token, err = tokenrequest.RequestTokenWithLocalCallback(o.Config, urlHandler, int(o.CallbackPort))
} else {
	token, err = tokenrequest.RequestTokenWithChallengeHandlers(o.Config, o.getAuthChallengeHandler())
}
if err != nil {
	return err
}
```